Task3/Project/DataControl/ViewModel/ApplicationViewModel.cs

```diff
@@ -26,120 +52,339 @@ public class ApplicationViewModel : INotifyPropertyChanged
+                messageWindow.ContentText = "Login can't be empty!";
+                messageWindow.ShowDialog();
+            }
+            if (string.IsNullOrWhiteSpace(password))
+            {
+                messageWindow.HeaderText = "Empty Password";
+                messageWindow.ContentText = "Password can't be empty!";
+                messageWindow.ShowDialog();
+            }
+            if (dataAccessService.LogIn(login, password))
+            {
+                currentDriver = dataAccessService.Driver;
+            }
+            else
+            {
+                messageWindow.HeaderText = "Account problem";
+                messageWindow.ContentText  = dataAccessService.Message;
+                messageWindow.ShowDialog();
+            }
         }
         private void LogOutMethod(object obj)
         {
-            throw new System.NotImplementedException();
+            currentDriver = null;
+            password = null;
+            login = null;
+            orders.Clear();
+            selectedOrder = null;
         }
         private void SignUpMethod(object obj)
         {
-            throw new System.NotImplementedException();
-        }
+            if (string.IsNullOrWhiteSpace(login))
+            {
+                messageWindow.HeaderText = "Empty Login";
+                messageWindow.ContentText = "Login can't be empty!";
+                messageWindow.ShowDialog();
+            }
 
+            if (string.IsNullOrWhiteSpace(password))
+            {
+                messageWindow.HeaderText = "Empty Password";
+                messageWindow.ContentText = "Password can't be empty!";
+                messageWindow.ShowDialog();
+            }
+            if (dataAccessService.SignUp(login, password))
+            {
+                currentDriver = dataAccessService.Driver;
+            }
+            else
+            {
+                messageWindow.HeaderText = "Account problem";
+                messageWindow.ContentText = dataAccessService.Message;
+                messageWindow.ShowDialog();
+            }
+        }
         private void StartGameMethod(object obj)
         {
-            throw new System.NotImplementedException();
+            if (!isSessionContinuing && currentDriver != null)
+            {
+                sessionTimer = new Timer(SESION_TIME);
+                sessionTimer.Elapsed += GameEnded;
+                sessionTimer.Start();
+                isSessionContinuing = true; 
+            }
         }
         private void ExecuteOrderMethod(object obj)
         {
-            throw new System.NotImplementedException();
+            progressWindow = new ProgressWindow(selectedOrder.Route.Time)
+            {
+                DataContext = this
+            };
+                progressWindow.ShowDialog();
+                if (progressWindow.DialogResult == true)
+                {
+                    currentScore += selectedOrder.Route.Price;

```

Comment:
use "CurrentScore" property and do not forget "OnPropertyChanged", so the changing will be showed on view